test/conformance/ingressv2/util.go

```diff
@@ -1070,15 +1070,22 @@ func WaitForHTTPRouteState(ctx context.Context, client *test.GatewayAPIClients,
 
 // IsHTTPRouteReady will check the status conditions of the ingress and return true if the ingress is
 // ready.
+// It returns false when
+// * one of ConditionRouteAdmitted is false.
+// * ConditionRouteAdmitted is not present. - eg. Ingress controller does not update the status in HTTPRoute yet.
 func IsHTTPRouteReady(r *gatewayv1alpha1.HTTPRoute) (bool, error) {
+	ready := false
```

Comment:
Oh I only now realised this is going over 2 lists. Then I think this still isn't **quite** right since this'll return `true` if one status is admitted and the other is non-existent. Frankly, I dunno if that can even happen so this might be useless. But I'll leave it anyway.

```golang
// IsHTTPRouteReady will check the status conditions of the ingress and return true if
// all gateways have been admitted.
func IsHTTPRouteReady(r *gatewayv1alpha1.HTTPRoute) (bool, error) {
	for _, gw := range r.Status.Gateways {
		if !isGatewayAdmitted(gw) {
			// Return false if _any_ of the gateways isn't admitted yet.
			return false, nil
		}
	}
	return true, nil
}

func isGatewayAdmitted(gw gatewayv1alpha1.RouteGatewayStatus) bool {
	for _, condition := range gw.Conditions {
		if condition.Type == string(gatewayv1alpha1.ConditionRouteAdmitted) {
			return condition.Status == metav1.ConditionTrue
		}
	}
	return false
}
```

That **should** have the right semantics (maybe we should add some unit tests?), if "right" means to return false if one of the Gateways is not admitted.

Reply:
Sounds good. Updated the code based on your suggestion.